### ISSUE TITLE
EB SSH tries to edit SGs to open port 22 even if it is already open

### ISSUE THREAD

**Issue by Likeyn** [aws/aws-elastic-beanstalk-cli#21]
### Description

The `eb ssh` command attempts to open port 22 in the target's SG even though it already successfully determined that it was open beforehand:
```
[...]
INFO: Attempting to open port 22.
2020-03-25 17:19:55,024 (DEBUG) ebcli.lib.aws : Making api call: (ec2, authorize_security_group_ingress) to region: us-east-1 with args:{'GroupId': 'sg-deadbeef', 'IpProtocol': 'tcp', 'ToPort': 22, 'FromPort': 22, 'CidrIp': '0.0.0.0/0'}
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : Response: {'Error': {'Code': 'InvalidPermission.Duplicate', 'Message': 'the specified rule "peer: 0.0.0.0/0, TCP, from port: 22, to port: 22, ALLOW" already exists'}, 'ResponseMetadata': {'RequestId': 'de-ad-b-ee-f', 'HTTPStatusCode': 400, 'date': 'Wed, 25 Mar 2020 16:19:54 GMT', 'RetryAttempts': 0}}
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : API call finished, status = 400
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : Received a 400 Error
INFO: SSH port 22 open.
[...]
```

The above error prevents you to SSH in (even if you actually CAN SSH in manually with the `ssh` command) if you don't have the permission to edit SGs, and if you do it is silently ignored and only shown in debug mode.

### Steps to reproduce

Run `eb ssh <eb-env> --debug` where `eb-env` is an EB environment which instances have a SG that already opens port 22.

### Observed result

The following error:
```
[...]
INFO: Attempting to open port 22.
2020-03-25 17:19:55,024 (DEBUG) ebcli.lib.aws : Making api call: (ec2, authorize_security_group_ingress) to region: us-east-1 with args:{'GroupId': 'sg-deadbeef', 'IpProtocol': 'tcp', 'ToPort': 22, 'FromPort': 22, 'CidrIp': '0.0.0.0/0'}
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : Response: {'Error': {'Code': 'InvalidPermission.Duplicate', 'Message': 'the specified rule "peer: 0.0.0.0/0, TCP, from port: 22, to port: 22, ALLOW" already exists'}, 'ResponseMetadata': {'RequestId': 'de-ad-b-ee-f', 'HTTPStatusCode': 400, 'date': 'Wed, 25 Mar 2020 16:19:54 GMT', 'RetryAttempts': 0}}
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : API call finished, status = 400
2020-03-25 17:19:55,279 (DEBUG) ebcli.lib.aws : Received a 400 Error
INFO: SSH port 22 open.
[...]
```
Plus, if you don't have permissions to edit SGs, you can't even attempt to SSH in.

### Expected result

No error, and you can attempt to SSH in even though you don't have permissions to edit SGs.

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

I was able to replicate on a WSL env and a OSX one.

**Comment by halcarleton** [comment]
Hey @Likeyn , thanks for reporting this. We will look into this and see what we can do to resolve the issue. Any updates will be reported back here.

**Comment by halcarleton** [comment]
Hey @Likeyn , could you clarify a few things for me? 

First, I have confirmed that a user that does not have `ec2:AuthorizeSecurityGroupIngress` permission for an environment's security group will experience a `NotAuthorizedError` when executing `eb ssh <env-name>` on that environment.

Is this what you are referring to when you say:
> The above error prevents you to SSH in (even if you actually CAN SSH in manually with the ssh command) if you don't have the permission to edit SGs

Second, could you expand on what you mean by:
> and if you do it is silently ignored and only shown in debug mode.

Are you saying that you are not able to ssh into your environment even when you _do_ have permission to `ec2:AuthorizeSecurityGroupIngress`? If so, what error is displayed?

Third, the logs you provided are expected, and should not prevent you from connecting to your instance. Could you add more detail about what behavior you are experiencing?

Thanks

**Comment by Likeyn** [comment]
Hey @halcarleton , sure.

What I mean is the security group already has port 22 opened (eg. permanently, by administrator's choice), and that the `ec2:AuthorizeSecurityGroupIngress` permission shouldn't be required to attempt to SSH in.

ATM it is the case, so even if EB CLI successfully determined that port 22 is already open for `0.0.0.0/0` in the security groups, a restricted user (understand with no `ec2:AuthorizeSecurityGroupIngress` permission) can **not** attempt to SSH in: EB CLI tries to open the port anyway, realises it can't because no permission, and blocks.

Expected behavior would be "*check if port 22 is open for `0.0.0.0/0`, if it is then **don't** attempt to open it and just try to SSH in straight away*".

I hope that helps

**Comment by shimaji** [comment]
Hi, I had exactly the same error in the exact same situation yesterday, but the error is gone after 12 hours and I can SSH using `eb ssh` without problems.

**Comment by halcarleton** [comment]
This should be fixed in [v3.18.2](https://pypi.org/project/awsebcli/3.18.2/)
